src/ray/gcs/gcs_server/gcs_resource_manager.h

```diff
@@ -180,6 +184,13 @@ class GcsResourceManager : public rpc::NodeResourceInfoHandler,
   /// Num of alive nodes in the cluster.
   size_t num_alive_nodes_ = 0;
 
+  /// Cluster resources constraints. This is the requested resources constraints manually
+  /// set by the user with autoscaler.sdk.request_resources().
+  /// If the the current cluster resources constraints are not satisfied, the autoscaler
+  /// is responsible to make sure the cluster has enough resources to fit the resources.
+  /// QQ: should we actually consider if this is the best API?
+  rpc::ClusterResourceConstraint cluster_resource_constraints_;
```

Comment:
We just need to save `ClusterResourceConstraint` as internal field for now since it will be overridden each time with `sdk.request_resources`